source/aws_imds_client.c

```diff
@@ -506,16 +499,30 @@ static void s_client_on_token_response(struct imds_user_data *user_data) {
      * token in following requests.
      */
     if (user_data->status_code != AWS_HTTP_STATUS_CODE_200_OK || user_data->current_result.len == 0) {
-        s_update_token_safely(user_data->client, NULL, false);
+        AWS_LOGF_DEBUG(
+            AWS_LS_IMDS_CLIENT,
+            "(id=%p) IMDS client failed to fetch token for requester %p, fall back to insecure path for the same "
```

Comment:
I am not sure if the `insure` terminology should be in our logs. Maybe we should use `v1` instead? In the future, if there is an imdsv3, this could be confusing.